src/steamship/agents/mixins/transports/slack.py

```diff
@@ -369,14 +397,22 @@ def _respond_to_block(self, incoming_message: Block):
         """Respond to a single inbound message from Slack, posting the response back to Slack."""
         try:
             chat_id = incoming_message.chat_id
-            context = self.agent_service.build_default_context(context_id=chat_id)
+            thread_ts = get_block_thread_ts(incoming_message)
+            # TODO (george) not sure this is always what we want, unfortunately.  Some bots may want to keep a
+            #  conversation going through threads, others may not?
+            context_id = chat_id if not thread_ts else f"{chat_id}-{thread_ts}"
```

Comment:
Possibly an advanced configuration option for this class.

Reply:
Agree that this could be a nice config option. 

Just thinking out:

- If the inbound message was already in a thread, it seems Always Right to respond in-thread.
- If the inbound message was not already in a thread, maybe that's when some kind of `always_thread_response` option would dictate?

Reply:
Oh, that's not even what I was meaning to discuss here but that's a good point and I will chase that.

The thing I was getting at here is that `build_default_context` IIRC will be used for recalling previous messages from a file using that ID, and I wanted to consider whether, if you're talking to the agent from within a thread, the requests to the bot from the outer channel should be considered.